src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx

```diff
@@ -28,6 +34,32 @@ export class DataSourceView extends React.Component<DataSourceViewProps, DataSou
     };
   }
 
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+  async componentDidMount() {
+    this._isMounted = true;
+    const selectedOption = this.props.selectedOption;
+    if (selectedOption && selectedOption.length === 1) {
+      if (selectedOption[0].id && !selectedOption[0].label) {
+        const title = (await getDataSourceById(selectedOption[0].id, this.props.savedObjectsClient))
+          .title;
+        if (!title) {
+          this.props.notifications.addWarning(
+            i18n.translate('dataSource.fetchDataSourceError', {
+              defaultMessage: `Invalid selectedOption: ${selectedOption[0].id}`,
```

Comment:
This is implementation error which makes more sense for developer, but here we need to show a toasts to user, which should be user friendly, something like `Data source with id ${dataSourceId} is not available` makes more sense.